### PR TITLE
triangle inequality for naturals and integers

### PR DESCRIPTION
**Pull request by lane-core**
On Dr. Escardo's invitation from mathstodon, I am contributing my proof of the triangle inequality in the Naturals and the Integers. In the Naturals, I define a new Absolute Difference operation and prove some elementary lemmas. I then add associated lemmas for Absolute Difference to the Natural Order lemmas, proving the triangle inequality on the way for the naturals under this operation. I then prove some useful lemmas for Absolute Value in the Integers related to this new operation, and conclude by proving the triangle inequality in the Integer Order file (for both abs and adding a translated lemma to absZ for convenience).

### CODE REVIEW

**Comment by martinescardo** [comment]
Thanks for your contribution. Will have a look now.

File: source/Integers/Abs.lagda [hunk 1]
```diff
@@ -169,3 +171,29 @@ abs-over-mult' (negsucc x) (negsucc y) = I
...
+\begin{code}
+
+pos-abs-is-absℤ : (x : ℤ) → pos (abs x) ＝ absℤ x
+pos-abs-is-absℤ (pos x) = refl
+pos-abs-is-absℤ (negsucc x) = refl
+
+abs-pos-plus-negsucc : (x y : ℕ) → abs (pos x +negsucc y) ＝ ∣ x - succ y ∣
+abs-pos-plus-negsucc zero y = ap abs (ℤ+-comm (pos 0) (negsucc y))
+abs-pos-plus-negsucc (succ x) zero = refl
+abs-pos-plus-negsucc (succ x) (succ y) =
```
Comment (martinescardo) [inline source/Integers/Abs.lagda:191]: I will approve this but will have to fix the indentation later. In `TypeTopology` the convention is that indentation is by one space character.

Comment (martinescardo) [reply source/Integers/Abs.lagda:191]: (Please fix all instances of that.)

File: source/Integers/Abs.lagda [hunk 2]
```diff
@@ -169,3 +171,29 @@ abs-over-mult' (negsucc x) (negsucc y) = I
...
+  abs (predℤ (pos (succ x) +negsucc y)) ＝⟨ i ⟩
+  abs (pos x +negsucc y) ＝⟨ abs-pos-plus-negsucc x y ⟩
+  ∣ x - succ y ∣ ∎
```
Comment (martinescardo) [inline source/Integers/Abs.lagda:194]: Please indent `∎` so that it shows below `＝⟨`.

File: source/Integers/Order.lagda [hunk 3]
```diff
@@ -786,3 +787,77 @@ Added by Todd
...
+    I : abs (pos y +negsucc x) ＝ abs (negsucc x +pos y)
+    I = ap abs (ℤ+-comm (pos y) (negsucc x))
+
+    II : succ (y ℕ+ x) ＝ succ x ℕ+ y
+    II = ap succ (addition-commutativity y x) ∙ succ-left x y ⁻¹
+
+triangle-inequality₃ : (x y : ℕ) → abs (negsucc x +negsucc y) ≤ℕ succ (succ x ℕ+ y)
+triangle-inequality₃ x y = transport (_≤ℕ succ (succ x ℕ+ y)) γ Γ
+  where
+  Γ : abs (pos (succ x) + pos (succ y)) ≤ℕ succ (succ x ℕ+ y)
```
Comment (martinescardo) [inline source/Integers/Order.lagda:830]: Indentation by 1 required (see comment above).

File: source/Integers/Order.lagda [hunk 4]
```diff
@@ -786,3 +787,77 @@ Added by Todd
...
+  Γ = triangle-inequality₀ (succ x) (succ y)
+
+  γ : abs (pos (succ x) + pos (succ y)) ＝ abs (negsucc x +negsucc y)
+  γ = abs (succℤ (pos (succ x) +pos y)) ＝⟨ i ⟩
```
Comment (martinescardo) [inline source/Integers/Order.lagda:834]: Indentation of the whole chain of equalities.

**Review by martinescardo — APPROVED** [review]
Nice! Thanks for your contribution. I will accept your pull request as it is, but please address the formatting problems in a second pull request after this.

**Comment by martinescardo** [comment]
Waiting for the checks to complete to merge.

**Comment by martinescardo** [comment]
You should also add yourself to the list of contributors in the top-level README file.

**Comment by martinescardo** [comment]
You have added the year of your contribution. Please also add the day and month.

**Comment by martinescardo** [comment]
When you have a chance, please to a pull request to address the above issues. Thanks.

**Comment by lane-core** [comment]
@martinescardo I will work on this clean-up over the weekend, thank you for accepting my work!

**Comment by lane-core** [comment]
Done with all these, I submitted a new PR to amend all these at the end of last week. Thank you!

**Pull request merged**